### PR TITLE
utils_test.qemu: enhancement for MemoryHotplugTest

### PR DESCRIPTION
**Pull request by luckyh**
Used new methods instead of 'check_memory'. These new methods firstly do
hotplug/hotunplug operation, and then verify the memory size change by
comparing change detected by guest os and the size of added/removed
memory device.

ID: 1392724

### CODE REVIEW

**Comment by luckyh** [comment]
@huangyum would you please help me review this? thanks!

**Comment by huangyum** [comment]
@luckyh  The check failed.

File: virttest/utils_test/qemu/__init__.py [hunk 1]
```diff
@@ -565,59 +576,113 @@ def update_vm_after_unplug(self, vm, dev):
...
-            if isinstance(dev, qdevices.Dimm):
-                addr = self.get_mem_addr(vm, dev.get_qid())
-                dev.set_param("addr", addr)
-                dev_type = "pc-dimm"
-            step = "Hotplug %s '%s' to VM" % (dev_type, dev.get_qid())
-            error_context.context(step, logging.info)
-            vm.devices.simple_hotplug(dev, vm.monitor)
-            self.update_vm_after_hotplug(vm, dev)
-        return devices
+        errmsg = "Could not find %s "
```
Comment (huangyum) [inline virttest/utils_test/qemu/__init__.py]: Missing something for %s ?

**Comment by luckyh** [comment]
Fixed

File: virttest/utils_test/qemu/__init__.py [hunk 2]
```diff
@@ -565,59 +576,113 @@ def update_vm_after_unplug(self, vm, dev):
...
+        :param vm: VM object
+        :param name: memory device name
+        """
+        step = "Hotplug memory device '%s'" % name
+        error_context.context(step, logging.info)
+        guest_old_msize = self.get_guest_total_mem(vm)
+        mem_dev_size = self.hotplug_memory(vm, name)
+
+        error_context.context("Verify memory info", logging.info)
+        added_mem_size = self.get_guest_total_mem(vm) - guest_old_msize
```
Comment (xutian) [inline virttest/utils_test/qemu/__init__.py]: In my mind, we can checked memory immediately in rhel7.0 guest, and we need to online it manually. And we also have case hotplug/unplug memory device when VM is paused status. This change may cause these test failed.

Thanks,
Xu

Comment (luckyh) [inline virttest/utils_test/qemu/__init__.py]: Let me test them.

Comment (luckyh) [inline virttest/utils_test/qemu/__init__.py]: Yes, the patch will makes some side effect.

**Comment by luckyh** [comment]
Close this since it will makes side effect.

**Pull request closed without merging**